gems/aws-sdk-core/spec/aws/client_spec.rb

```diff
@@ -150,6 +150,22 @@ module Aws
         expect(client.example_operation.string).to eq('value')
       end
 
+      it 'allows api requests to be logged when stubbed' do 
+        client = client_class.new(stub_responses: {
+          example_operation: { string: 'value' }
+        })
+        expect(client.api_requests.length).to eq(0)
+        client.example_operation
+        expect(client.api_requests.length).to eq(1)
```

Comment:
Let's actually verify the contents, and use an example operation with parameters to verify that users can do the same validations on a first-class basis as you can in the aws-record examples. This is especially important because if we make changes later that would break stubbing behavior, we want to exercise all of it (assertions on everything we expose) so that those changes would break the test. Right now, the test has no contract on what actually is in the api_requests object.